File: packages/health_connector_hk_ios/ios/health_connector_hk_ios/Sources/health_connector_hk_ios/mappers/measurement_unit_mappers/TemperatureMapper.swift
import Foundation
import HealthKit

extension TemperatureDto {
    /// Converts this DTO to a HealthKit `HKQuantity`.
    func toHealthKit() -> HKQuantity {
        HKQuantity(unit: .degreeCelsius(), doubleValue: celsius)
    }
}

extension HKQuantity {
    /// Converts this HealthKit quantity to a `TemperatureDto`.
    ///
    /// Uses celsius as the transfer unit for consistency.
    func toTemperatureDto() -> TemperatureDto {
        let celsius = doubleValue(for: .degreeCelsius())
        return TemperatureDto(celsius: celsius)
    }
}

File: packages/health_connector_hk_ios/lib/src/mappers/measurement_unit_mappers/temperature_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Temperature, sinceV1_0_0;
import 'package:health_connector_hk_ios/src/pigeon/health_connector_hk_ios_api.g.dart'
    show TemperatureDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Temperature] to [TemperatureDto].
@sinceV1_0_0
@internal
/// Converts [Temperature] to [TemperatureDto].
@sinceV1_0_0
@internal
extension TemperatureToDto on Temperature {
  TemperatureDto toDto() {
    // Uses celsius as the transfer unit for consistency.
    return TemperatureDto(celsius: inCelsius);
  }
}

/// Converts [TemperatureDto] to [Temperature].
@sinceV1_0_0
@internal
/// Converts [TemperatureDto] to [Temperature].
@sinceV1_0_0
@internal
extension TemperatureDtoToDomain on TemperatureDto {
  Temperature toDomain() {
    return Temperature.celsius(celsius);
  }
}
